### DISCUSSION TITLE
Hue emulation does not send M-SEARCH response (Sonoff ZbBridge)

### DISCUSSION THREAD

**Discussion by Barracuda09** [arendst/Tasmota#10950] [Ideas]
There is an `"` between `</specVersion>"<URLBase>` which will result in an ill formed xml file. Therefore the discovery will fail most likely. This is introduced in #9505 

https://github.com/arendst/Tasmota/blob/d3d4a49e7f0824e31e53fa2f2f757a9a71a9edc0/tasmota/xdrv_20_hue.ino#L233-L240

Edit:
. |  .
--|--
Program Version | 9.2.0.4(zbbridge)
Build Date & Time | 2021-02-01T16:48:15
Core/SDK Version | 2_7_4_9/2.2.2-dev(38a443e)

**Comment by s-hadinger** [comment]
Good catch. I will fix it tomorrow.

**Comment by Barracuda09** [comment]
@s-hadinger 

I see you fixed the ill formed description.xml in current build:
. | .
-- | --
Program Version | 9.2.0.7(zbbridge)
Build Date & Time|2021-02-14T17:44:46
Core/SDK Version|2_7_4_9/2.2.2-dev(38a443e)

But this still does not fix the issue with the discovery by _Alexa_ or _Windows Network Scan_ **BUT** good news is in build 9.2.0 it works. So back to the drawing board, and find out what changed more in the maintime.

Edit, so for good order this one works:

. | .
-- | --
Program Version | 9.2.0(zbbridge)
Build Date & Time|2020-12-21T15:06:32
Build Date & Time|2_7_4_9/2.2.2-dev(38a443e)

**Comment by Barracuda09** [comment]
@s-hadinger 

Probably here is the discovery problem for _Alexa (Hue)_ and _Windows Network Scan_
https://github.com/arendst/Tasmota/blob/725650cac50febeec8b404a1ec89171770cac705/tasmota/support_udp.ino#L137-L145

This has changed with this #10564 -> The conditions are now the same and previous it was not. 

I don't know what _devices_present_ does or imply

**Reply by s-hadinger** [reply]
You are right. This is a bad copy paste. 

The response wouldn't be sent if there wasn't already a relay or pwm. I will fix this asqp
